### PR TITLE
Sreq

### PR DESCRIPTION
**Pull request by RanjiniM**
First round of addressing code review comments. The following are pending still

1. Move methods out of DbContext into Repository
2. Write tests for the feature
3. Mapping Admins table in Support Request DB to Admins in NuGetGallery

### CODE REVIEW

File: src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+                if (newIssueStatusName != -1)
+                {
+                    currentIssue.IssueStatus = newIssueStatusName;
+                }
+                issue.SaveChanges();
+               
+                issue.AddHistoryEntry(currentIssue, GetLoggedInUser());
+            }
+
+            return RedirectToAction("index", new
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs]: **I**ndex (capital `I`) - several other occurrences as well.

File: src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+            {
+                passedInAssignedToFilter = filterResultsView.AssignedTo,
+                passedInIssueStatusNameFilter = filterResultsView.IssueStatusName,
+                passedInReasonFilter = filterResultsView.Reason,
+                passedInStatusId = filterResultsView.StatusID,
+                passedInPageNumber = filterResultsView.PageNumber
+            });
+        }
+       
+        public ActionResult index(int passedInPageNumber = 0,
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs]: **I**ndex (capital `I`)

File: src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+            int passedInStatusId = 1,
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs]: Is there a reason to call these `passedIn` ? Would just name them `statusId`, `assignedToFilter`, ...

File: src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+            if (count % PageSize == 0)
+            {
+                ViewBag.MaxPage = count / PageSize;
+            }
+            else
+            {
+                ViewBag.MaxPage = count / PageSize + 1;
+            }
+
+            ViewBag.Page = passedInPageNumber;
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs]: Paging could be part of the model as well (for example `PackageListViewModel` does this as well). Not a hard requirement but would be good for code base consistency.

File: src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs [hunk 5]
```diff
@@ -0,0 +1,443 @@
...
+                passedInReasonFilter = indexViewModel.Filter.Reason,
+                passedInStatusId = indexViewModel.Filter.StatusID,
+                passedInPageNumber = indexViewModel.Filter.PageNumber
+            });
+        }
+
+        public ActionResult Edit(int id = 1,
+            int passedInPageNumber = 0, int passedInStatusId = 1,
+            int passedInAssignedToFilter = -1, int passedInIssueStatusNameFilter = -1,
+            string passedInReasonFilter = "[Reason]")
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs]: Can the `[Reason]` become a constant?

File: src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs [hunk 6]
```diff
@@ -0,0 +1,443 @@
...
+            var filteredIssueViews = new List<IssueViewModel>();
+
+            var admins = admin.GetAllAdmins();
+            var issueStatuses = issueStatus.GetAllIssueStatuses();
+            var adminsList = GetListOfAdmins(admins, -1);
+            var issueStatusesList = GetListOfIssueStatuses(issueStatuses, -1);
+
+            foreach (Issue i in filteredIssues)
+            {
+                var rv = new IssueViewModel();
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs]: Nit: Can you give this a more meaningful name?

File: src/NuGetGallery/Areas/Admin/Models/IssueModel.cs [hunk 7]
```diff
@@ -0,0 +1,173 @@
+namespace NuGetGallery.Areas.Admin.Models
+{ 
+    using System;
+    using System.Data.Entity;
+    using System.ComponentModel.DataAnnotations.Schema;
+    using System.Linq;
+    using System.Collections.Generic;
+
+    public partial class IssueModel : DbContext
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Models/IssueModel.cs]: Why all the different models? Looking at the methods, these should be repository services. E.g. `GetOpenIssues` is typical logic for the repository. Moving such methods to a repository, testing becomes easier as there's no need to "fake" the `DbContext` and database connection during tests. A simple test repository can be created based on a collection, and unit tests can then use that simple implementation.

File: src/NuGetGallery/Views/Shared/Layout.cshtml [hunk 8]
```diff
@@ -59,5 +59,9 @@
         </div>
         @Scripts.Render("~/Scripts/all")
         @RenderSection("BottomScripts", required: false)
+
+        @Scripts.Render("~/bundles/jquery")
```
Comment (maartenba) [inline src/NuGetGallery/Views/Shared/Layout.cshtml]: Great to have these factored out into separate bundles! Would it be an option to only render them on the support request pages and not on regular gallery pages? (smaller page payload, faster pages)

File: src/NuGetGallery/Web.config [hunk 9]
```diff
@@ -244,7 +250,7 @@
       <error statusCode="404" redirect="~/Errors/404" />
       <error statusCode="500" redirect="~/Errors/500" />
     </customErrors>
-    <sessionState mode="Off" />
+    <!--<sessionState mode="Off" />-->
```
Comment (maartenba) [inline src/NuGetGallery/Web.config]: Can they be turned off?

**Comment by maartenba** [comment]
Can you rebase on top of the `develop` branch? There are a few changes in there you may want to have in your branch as well.

File: .vs/config/applicationhost.config [hunk 10]
```diff
@@ -155,7 +155,7 @@
         <sites>
             <site name="NuGet Gallery (nuget.localtest.me)" id="2">
                 <application path="/">
-                    <virtualDirectory path="/" physicalPath="..\..\src\NuGetGallery" />
+                    <virtualDirectory path="/" physicalPath="C:\Users\ranjinim\Documents\GitHub\NuGetGallery\src\NuGetGallery" />
```
Comment (xavierdecoster) [inline .vs/config/applicationhost.config]: this change should not be committed

Comment (maartenba) [inline .vs/config/applicationhost.config]: This is one that is gitignored in `develop`, so rebasing will fix this.

File: src/NuGetGallery/Areas/Admin/Views/Request/Edit.cshtml [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+            <div class="panel-body">
+                <ul class="list-group">
+                    <li class="list-group-item" style="">
+                        <div class="row">
+                            <span>@Html.DisplayFor(model => model.Issue.Details)</span>
+                        </div>
+                        <div class="row">
+                            <div class="col-md-10">
+
+                                <span class="label" style="background-color: #CCCCCC; color: #000000">@Html.DisplayFor(model => model.Issue.CreatedBy)</span>
```
Comment (xavierdecoster) [inline src/NuGetGallery/Areas/Admin/Views/Request/Edit.cshtml]: preferably extract inline styles into css stylesheets

File: src/NuGetGallery/Content/Layout.css [hunk 12]
```diff
@@ -141,7 +141,7 @@ nav.main {
 
 #searchBox {
     background-color: #fff;
-    border: solid 1px #05005F;
+    /*border: solid 1px #05005F;*/
```
Comment (xavierdecoster) [inline src/NuGetGallery/Content/Layout.css]: nit: remove instead of commenting, reduces response payload a (tiny) bit

Comment (maartenba) [inline src/NuGetGallery/Content/Layout.css]: Also why is it removed in the first place, it makes the text box look different.

File: src/NuGetGallery/Controllers/PackagesController.cs [hunk 13]
```diff
@@ -404,6 +408,125 @@ public virtual ActionResult UndoPendingEdits(string id, string version)
...
+            {
+                //swallow
+            }
+        }
+
+        private void TriggerAPagerDutyIncident(string errorMeesage)
+        {
+            try
+            {
+                var pagerDutyAPIKey = WebConfigurationManager.AppSettings["Gallery.PagerDutyAPIKey"];
```
Comment (xavierdecoster) [inline src/NuGetGallery/Controllers/PackagesController.cs]: should we cache these config values?

**Comment by maartenba** [comment]
Can this PR be closed?

**Pull request closed without merging**